packages/compiler-core/src/transforms/vBind.ts

```diff
@@ -98,6 +83,32 @@ export const transformBind: DirectiveTransform = (dir, _node, context) => {
   }
 }
 
+export const transformBindShorthand = (
+  dir: DirectiveNode,
+  context: TransformContext,
+) => {
+  const arg = dir.arg!
+  const { loc } = dir
+  if (arg.type !== NodeTypes.SIMPLE_EXPRESSION || !arg.isStatic) {
+    // only simple expression is allowed for same-name shorthand
+    context.onError(
+      createCompilerError(
+        ErrorCodes.X_V_BIND_INVALID_SAME_NAME_ARGUMENT,
+        arg.loc,
+      ),
+    )
+    return {
+      props: [createObjectProperty(arg, createSimpleExpression('', true, loc))],
+    }
+  }
+
+  const propName = camelize((arg as SimpleExpressionNode).content)
+  dir.exp = createSimpleExpression(propName, false, arg.loc)
+  if (!__BROWSER__) {
+    dir.exp = processExpression(dir.exp, context)
+  }
+}
```

Comment:
This function seems to have two modes of operation. In one branch it returns something, in the other branch it mutates something. That seems like a bit of a red flag.

The function is called in two places, but the return value is only considered in one of those calls. That also seems like a red flag.

Perhaps it's worth reconsidering the responsibilities of this function. Doesn't the `findProp` call in `vFor.ts` avoid the error case? Does the error case need to be handled inside the function, or could it be left where it was previously?

Reply:
Thank you for your suggestion. You make a valid point. I agree with refactoring the function to adhere to the Single Responsibility Principle. Separating concerns will improve the code's maintainability and readability. Regarding the conditional statement, I agree that it should be returned to its previous location. Your feedback has provided valuable insights, and I appreciate your contribution to enhancing the codebase's quality.